### PR TITLE
MSDynamics: Base Mapping Component

### PR DESCRIPTION
**Pull request by anishfyle**
## Summary by CodeRabbit

- **New Features**
  - Implemented a new mapping interface for business integrations, enhancing user interaction and data handling.
  
- **Enhancements**
  - Added conditional loading indicators to improve user experience during data processing.
  - Introduced the ability to automatically map employee fields, streamlining the setup process.

- **Refactor**
  - Refactored component initialization to include new services and dependencies for extended functionality.

### CODE REVIEW

**Comment by anishfyle** [comment]
@coderabbitai review

File: src/app/integrations/business-central/business-central-main/business-central-mapping/business-central-base-mapping/business-central-base-mapping.component.ts [hunk 1]
```diff
@@ -7,9 +12,82 @@ import { Component, OnInit } from '@angular/core';
...
+
+  isLoading: boolean = true;
+
+  showAutoMapEmployee: boolean = false;
+
+  reimbursableExpenseObject: string;
+
+  cccExpenseObject: string;
+
+  employeeFieldMapping: FyleField = FyleField.VENDOR;
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-main/business-central-mapping/business-central-base-mapping/business-central-base-mapping.component.ts:27]: this is not fixed, check QBO for example

File: src/app/integrations/business-central/business-central-main/business-central-mapping/business-central-base-mapping/business-central-base-mapping.component.ts [hunk 2]
```diff
@@ -7,9 +12,82 @@ import { Component, OnInit } from '@angular/core';
...
+  getSourceType() {
+    if (this.sourceField==='EMPLOYEE') {
+      return 'VENDOR';
+    }
+
+    if (this.sourceField==='CATEGORY') {
+      return 'ACCOUNT';
+    }
+
+    return '';
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-main/business-central-mapping/business-central-base-mapping/business-central-base-mapping.component.ts:57]: check QBO for this function, needs few changes

File: src/app/integrations/business-central/business-central-main/business-central-mapping/business-central-base-mapping/business-central-base-mapping.component.ts [hunk 3]
```diff
@@ -7,9 +12,82 @@ import { Component, OnInit } from '@angular/core';
...
+        if (this.sourceField===FyleField.EMPLOYEE) {
+          this.destinationOptions = this.destinationField===FyleField.EMPLOYEE ? response.EMPLOYEE : response.VENDOR;
+        }
+        if (this.sourceField==='CATEGORY') {
+          if (this.destinationField === 'EXPENSE_TYPE') {
+            this.destinationOptions = response.EXPENSE_TYPE;
+          } else {
+            this.destinationOptions = response.ACCOUNT;
+          }
+        }
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-main/business-central-mapping/business-central-base-mapping/business-central-base-mapping.component.ts:79]: check QBO for example on how to assign destinationOptions

**Comment by anishfyle** [comment]
@coderabbitai review

File: src/app/integrations/business-central/business-central-main/business-central-mapping/business-central-base-mapping/business-central-base-mapping.component.ts [hunk 4]
```diff
@@ -7,9 +16,78 @@ import { Component, OnInit } from '@angular/core';
...
+
+  isLoading: boolean = true;
+
+  showAutoMapEmployee: boolean = false;
+
+  reimbursableExpenseObject: string;
+
+  cccExpenseObject: string;
+
+  employeeFieldMapping: FyleField = FyleField.VENDOR;
```
Comment (ashwin1111) [inline src/app/integrations/business-central/business-central-main/business-central-mapping/business-central-base-mapping/business-central-base-mapping.component.ts:31]: remove default value

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**